webapi/src/test/scala/org/knora/webapi/slice/admin/repo/service/KnoraUserGroupRepoLiveSpec.scala

```diff
@@ -21,3 +28,29 @@ object KnoraUserGroupRepoInMemory {
   val layer = ZLayer.fromZIO(Ref.make(List.empty[KnoraUserGroup])) >>>
     ZLayer.derive[KnoraUserGroupRepoInMemory]
 }
+
+object KnoraUserGroupRepoLiveSpec extends ZIOSpecDefault {
+  val spec: Spec[Any, Any] = suite("KnoraUserGroupRepoLive")(
+    suite("findById")(
+      test("findById given a non existing user should return None") {
+        ZIO.serviceWithZIO[KnoraUserGroupRepo](userGroupRepo =>
+          for {
+            userGroup <- userGroupRepo.findById(GroupIri.unsafeFrom("http://rdfh.ch/groups/0001/1234"))
+          } yield assertTrue(userGroup.isEmpty)
+        )
+      },
+      test("findById given an existing user should return that user") {
+        ZIO.serviceWithZIO[KnoraUserGroupRepo](userGroupRepo =>
+          for {
+            _         <- userGroupRepo.save(testUserGroup)
+            userGroup <- userGroupRepo.findById(testUserGroup.id)
+          } yield assertTrue(userGroup.contains(testUserGroup))
+        )
+      }
```

Comment:
**comment (non-blocking):** if you wanted to test `findById` in separation, with out relying on `save`, you could also get the `TriplestoreServiceInMemory` from the environment and add data directly to the triplestore. (There should be an example of it in the KnoraProject specs somewhere)

Reply:
True, I also had this in mind, that's why I checked how to get all the statements with `datasetStatements` in the `TriplestoreServiceInMemory`, however, I'll leave this as is for now.